sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/rest_client.hpp

```diff
@@ -1091,6 +1091,23 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
         std::string NextMarker;
         Nullable<std::string> DirectoryId;
       };
+      /**
+       * @brief Access rights of the access policy.
+       */
+      class AccessRight final {
+      public:
+        AccessRight() = default;
+        explicit AccessRight(std::string value) : m_value(std::move(value)) {}
+        bool operator==(const AccessRight& other) const { return m_value == other.m_value; }
```

Comment:
Use an `Azure::Core::_internal::ExtendableEnumeration` rather than rolling your own definition. At a minimum, it reduces the amount of boilerplate associated with extendable enumerations, at best it provides a consistent implementation for them (allowing for centralized improvements for functionality).

Reply:
This is machine generated code in _detail namespace and invisible to customers. In convenience layer, it will be a extendable enumeration which support ` | `, ` & `, ` ^ `